### DISCUSSION TITLE
C++11 currently something newer would be useful?

### DISCUSSION THREAD

**Discussion by cb88** [chipKIT32/chipKIT-core#487] [General]
I believe Chipkit is currently C++11? It sure would be nice to be able to do some fancier constexpr compile time stuff in Chipkit using C++14 or 20. Consexpr in C++11 only allow you to write functions with a return statement and nothing else, but later C++ versions allow fairly complex evaluations and multiple returns statements so long as the result can be determined at compile time. 

It would save me a few thousand bytes in one of my current chipkit programs if I were able to do a constexpr date conversion at compile time.

Dunno how much effort that would be though, I know I've seen mentioned that chipkit has a few bits changed in the compiler.

**Comment by JacobChrist** [comment]
If there is any hope @majenkotech would be the wonder.  Microchip used to maintain the compiler for chipKIT but they dropped support shortly after buying Atmel.

**Reply by cb88** [reply]
In the meantime, I found someone had done the preprocessor voodoo to make it work that way also... 
https://stackoverflow.com/questions/1765014/convert-string-from-date-into-a-time-t
